sdk/docs/iot/README.md

```diff
@@ -179,160 +179,37 @@ void my_property_func()
 Telemetry functionality can be achieved by sending a user payload to a specific topic. In order to get the appropriate topic to which to send, use the `az_iot_hub_client_telemetry_get_publish_topic()` API. An example use case is below.
 
 ```C
-//FOR SIMPLICITY THIS DOES NOT HAVE ERROR CHECKING. IN PRODUCTION ENSURE PROPER ERROR CHECKING.
+// FOR SIMPLICITY THIS DOES NOT HAVE ERROR CHECKING. IN PRODUCTION ENSURE PROPER ERROR CHECKING.
 
 static az_iot_hub_client my_client;
 static az_span my_iothub_hostname = AZ_SPAN_LITERAL_FROM_STR("contoso.azure-devices.net");
 static az_span my_device_id = AZ_SPAN_LITERAL_FROM_STR("contoso_device");
 
-void my_telemetry_func()
+void my_telemetry_func(void)
 {
-  //Initialize the client to then pass to the telemetry API
+  // Initialize the client to then pass to the telemetry API.
   az_iot_hub_client_init(&my_client, my_iothub_hostname, my_device_id, NULL);
 
-  //Allocate a char buffer with capacity large enough to put the telemetry topic.
+  // Allocate a char buffer with capacity large enough to put the telemetry topic.
   char telemetry_topic[64];
   size_t telemetry_topic_length;
 
-  //Get the NULL terminated topic and put in telemetry_topic to send the telemetry
-  az_iot_hub_client_telemetry_get_publish_topic(&my_client, NULL, telemetry_topic,
-                                    sizeof(telemetry_topic), &telemetry_topic_length);
-}
-```
-
-### IoT Hub Client with MQTT Stack
```

Comment:
I'm going to push back on this one. Why is this being removed? I think John's original comment is that the readme is a little long but that was attempted to be solved with the table of contents.

Reply:
For two reasons, 1, John's comment about the readme being too long, and 2, because this code is accomplished in the samples.  I see it attempting to be MQTT stack agnostic here, but I actually found that to be more confusing since the mqtt function calls appeared to be hypothetical?  I would prefer to direct the user to the samples at this point.

Reply:
I have added a coding_patterns.md file that now houses this.